From 0a7997638ce0541aa54d5ec411c4445380e05699 Mon Sep 17 00:00:00 2001
From: Yu Liao <liaoyu15@huawei.com>
Date: Mon, 3 Apr 2023 15:25:03 +0800
Subject: [PATCH 0029/1646] [Huawei] timekeeping: Make
 CLOCKSOURCE_VALIDATE_LAST_CYCLE configurable

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188626

-------------------------------

Currently, only x86 architecture supports the CLOCKSOURCE_VALIDATE_LAST_CYCLE
option. This option ensures that the timestamps returned by the clocksource are
monotonically increasing, and helps avoid issues caused by hardware failures.

This commit makes CLOCKSOURCE_VALIDATE_LAST_CYCLE configurable on
the arm64 architecture, helps increase system stability and reliability.

Signed-off-by: Yu Liao <liaoyu15@huawei.com>
---
 kernel/time/Kconfig | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/kernel/time/Kconfig b/kernel/time/Kconfig
index a09b1d61df6a..0572a2fb6b12 100644
--- a/kernel/time/Kconfig
+++ b/kernel/time/Kconfig
@@ -17,11 +17,6 @@ config ARCH_CLOCKSOURCE_DATA
 config ARCH_CLOCKSOURCE_INIT
 	bool
 
-# Clocksources require validation of the clocksource against the last
-# cycle update - x86/TSC misfeature
-config CLOCKSOURCE_VALIDATE_LAST_CYCLE
-	bool
-
 # Timekeeping vsyscall support
 config GENERIC_TIME_VSYSCALL
 	bool
@@ -173,5 +168,13 @@ config HIGH_RES_TIMERS
 	  hardware is not capable then this option only increases
 	  the size of the kernel image.
 
+# Clocksources require validation of the clocksource against the last
+# cycle update - x86/TSC misfeature
+config CLOCKSOURCE_VALIDATE_LAST_CYCLE
+	bool "Validate last cycle of clocksource"
+	help
+	  Enable this option to validate the last cycle to prevent issues
+	  caused by clocksource going backwards.
+
 endmenu
 endif
-- 
2.17.1

